aws-auth-cognito/src/main/java/com/amplifyframework/auth/cognito/AWSCognitoAuthPlugin.kt

```diff
@@ -639,7 +641,61 @@ class AWSCognitoAuthPlugin : AuthPlugin<AWSCognitoAuthServiceBehavior>() {
         onSuccess: Consumer<MutableList<AuthDevice>>,
         onError: Consumer<AuthException>
     ) {
-        TODO("Not yet implemented")
+        var token: StateChangeListenerToken? = null
+        token = credentialStoreStateMachine.listen(
+            {
+                when (it) {
+                    is CredentialStoreState.Success -> {
+                        token?.let(credentialStoreStateMachine::cancel)
+                        val accessToken = it.storedCredentials?.cognitoUserPoolTokens?.accessToken
+                        if (!accessToken.isNullOrEmpty()) {
+                            runBlocking {
+                                _fetchDevices(
+                                    accessToken,
+                                    onSuccess,
+                                    onError
+                                )
+                            }
+                        } else {
+                            onError.accept(AuthException.InvalidStateException())
+                        }
+                    }
+                    is CredentialStoreState.Error -> {
+                        token?.let(credentialStoreStateMachine::cancel)
+                        onError.accept(AuthException.InvalidStateException())
+                    }
+                    else -> {
+                        // no-op
+                    }
+                }
+            },
+            {
+                credentialStoreStateMachine.send(
+                    CredentialStoreEvent(CredentialStoreEvent.EventType.LoadCredentialStore())
+                )
+            }
+        )
+    }
+
+    private suspend fun _fetchDevices(
+        token: String, onSuccess: Consumer<MutableList<AuthDevice>>,
+        onError: Consumer<AuthException>
+    ) {
+        val cognitoIdentityProviderClient = configuration.userPool?.let { it ->
+            CognitoIdentityProviderClient { this.region = it.region }
+        }
+        try {
+            val response = cognitoIdentityProviderClient?.listDevices(ListDevicesRequest.invoke { accessToken = token })
```

Comment:
User needs to be signed In to make this call. Use auth state machines current state to enforce this.